core/includes/customizer/helper.php

```diff
@@ -814,10 +814,24 @@ function responsive_custom_excerpt_length( $length ) {
 function responsive_custom_excerpt_limit_text( $text ) {
 	$limit = get_theme_mod( 'responsive_excerpt_length' );
 	if ( ! empty( $limit ) ) {
-		if ( str_word_count( $text, 0 ) > $limit ) {
-			$words = str_word_count( $text, 2 );
-			$pos   = array_keys( $words );
-			$text  = substr( $text, 0, $pos[ $limit ] );
+		$text      = wp_strip_all_tags( $text );
+		$text      = strip_shortcodes( $text );
+		$check_end = 0;
+		if ( $limit >= str_word_count( $text, 0 ) - 1 ) {
+			$check_end        = 1;
+			$limit            = str_word_count( $text, 0 ) - 1;
+			$words            = str_word_count( $text, 2 );
+			$pos              = array_keys( $words );
+			$excerpt_end_text = get_theme_mod( 'responsive_blog_read_more_text', __( 'Read more &raquo;', 'responsive' ) );
+			$excerpt_end_arr  = explode( ' ', $excerpt_end_text );
+			$occ_char         = strripos( $text, $excerpt_end_arr[0] );
+			$text             = substr( $text, 0, $pos[ $limit ] );
+			$text_two         = substr( '' . $text, 0, $occ_char );
```

Comment:
@SachinLagad345 I can't understand the code, what is $text_two, $occ_char? can you please use meaningfull variable names?